sunpy/data/manager/manager.py

```diff
@@ -0,0 +1,135 @@
+import pathlib
+import functools
+from contextlib import contextmanager
+
+from sunpy.util.util import hash_file
+
+
+class DataManager:
+    """
+    DataManager
+    """
+
+    def __init__(self, cache):
+        # TODO: Folder prefix should be init argument
+        self._cache = cache
+
+        self._file_cache = {}
+
+        self._skip_hash_check = False
+        self._skip_file = {}  # Dict[str, str]
+
+    def require(self, name, urls, sha_hash):
+        """decorator for doing stuff
```

Comment:
```suggestion
        """
        Decorator for doing stuff
```

Should fill in the rest of that sentence.